modules/urlsigning-service-impl/src/main/java/org/opencastproject/security/urlsigning/provider/impl/AbstractUrlSigningProvider.java

```diff
@@ -101,24 +109,32 @@ public void setSecurityService(SecurityService securityService) {
    * @return The current set of url beginnings this signing provider is looking for.
    */
   public Set<String> getUris() {
-    return keys.values().stream()
-            .map(keyEntry -> keyEntry.url)
-            .collect(Collectors.collectingAndThen(
-                    Collectors.toSet(),
-                    Collections::unmodifiableSet));
+    return Collections.unmodifiableSet(urls.keySet());
   }
 
   /**
-   * If available get a {@link KeyEntry} if there is a matching Url matcher.
+   * If available get a {@link Key} if there is a matching Url matcher.
    *
    * @param baseUrl
    *          The url to check against the possible matchers.
-   * @return The {@link KeyEntry} if it is available.
+   * @return The {@link Key} if it is available.
    */
-  private Optional<Map.Entry<String, KeyEntry>> getKeyEntry(String baseUrl) {
-    return keys.entrySet().stream()
-            .filter(entry -> baseUrl.startsWith(entry.getValue().url))
-            .findAny();
+  private Key getKey(String baseUrl) {
```

Comment:
There is simply no reason why we should pack the answer into a wrapper object.

Reply:
I disagree. Expressing the exact type of object a method can return as explicitly as possible is very useful to force consumers to think about all necessary (edge) cases. This method returns `null` sometimes, which every consumer needs to consider. That's the main purpose of things like `Optional` in languages like Java.

Now, maybe it's justified to remove this overhead here, this being a PR about performance and all. Also this is a `private` method, so the effects of being less explicit are localized at least, but I still maintain that `Optional` could help when working on the module itself. Also I think that this might be a bit of a premature optimization.

But okay, I won't block the PR for this. Just wanted to spread some :heart: for type safety.